doc/build.md

```diff
@@ -1,35 +1,31 @@
-# Build instructions for Linux
+# Build instructions
 
-See also [Istio Development](https://github.com/istio/istio/blob/master/devel/README.md) for setup and general tips.
```

Comment:
I wonder if there is a way to improve the general document rather than diverge between istio repos

I would in theory remove this doc entirely in favor of the the shared doc ?

Reply:
@ldemailly @rshriram @kyessenov Please see the shortened version of the build instructions. I just referenced the general document and stated the differences of Pilot.

What I learned - there is duplication of build artifacts between pilot and mixer. Both have Makefiles which are slightly different and both have `bin/bazel_to_go.py` which are different as well. I would propose to consider creating an `istio/build` repositiory and putting Makefile and scripts there. Then this repository can be shared between Istio repositories as git submodule or in some other way, so all the Istio repositories will use the same Makefiles and scripts, and the development guide will be the same for all.

Reply:
and we could just build off the istio/build ;)

Reply:
https://github.com/istio/test-infra/issues/91